sdk/test-utils/perf/src/program.ts

```diff
@@ -1,286 +1,58 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-import { AbortController } from "@azure/abort-controller";
-import { PerfTest } from "./perfTest";
-import {
-  parsePerfOption,
-  defaultPerfOptions,
-  DefaultPerfOptions,
-  ParsedPerfOptions,
-} from "./options";
-import { PerfParallel } from "./parallel";
-import { exec } from "child_process";
-import { formatDuration } from "./utils";
-import { PerfTestBase, PerfTestConstructor } from "./perfTestBase";
-
-/**
- * PerfProgram
- * receives a class extending PerfTest with specific command line parameter names (or just "string"),
- * then gets the test ready for a performance/stress test run.
- *
- * Use it like:
- *
- * ```ts
- * export class Delay500ms extends PerfTest {
- *   public options = {};
- *   async run(): Promise<void> {
- *     await delay(500);
- *   }
- * }
- *
- * const perfProgram = new PerfProgram(Delay500ms);
- *
- * perfProgram.run();
- * ```
- */
-export class PerfProgram {
-  private testName: string;
-  private parsedDefaultOptions: Required<ParsedPerfOptions<DefaultPerfOptions>>;
-  private parallelNumber: number;
-  private tests: PerfTestBase[];
+import { PerfTestConstructor } from "./perfTestBase";
+import minimist, { ParsedArgs } from "minimist";
+import { WorkerPerfProgram } from "./workerProgram";
+import { ManagerPerfProgram } from "./managerProgram";
+import { multicoreUtils } from "./multicore";
+
+export interface PerfProgram {
+  run(): Promise<void>;
+}
 
-  /**
-   * Receives a test class to instantiate and execute.
-   * Parses the test's options and creates some shortcuts.
-   * It will instantiate the same test class as many times as the "parallel" command line parameter specifies,
-   * which defaults to 1.
-   *
-   * @param testClass The testClass to be instantiated.
-   */
-  constructor(testClass: PerfTestConstructor) {
-    this.testName = testClass.name;
-    this.parsedDefaultOptions = parsePerfOption(defaultPerfOptions);
-    this.parallelNumber = Number(this.parsedDefaultOptions.parallel.value);
+export function createPerfProgram(testClass: PerfTestConstructor): PerfProgram;
+export function createPerfProgram(testClasses: PerfTestConstructor[]): PerfProgram;
 
-    console.log(`=== Creating ${this.parallelNumber} instance(s) of ${this.testName} ===`);
-    this.tests = new Array<PerfTest<DefaultPerfOptions>>(this.parallelNumber);
+export function createPerfProgram(
+  testClassOrClasses: PerfTestConstructor | PerfTestConstructor[]
```

Comment:
I wonder if this should just be `export function createPerfProgram(...testClasses: PerfTestConstructor[])` to simplify the logic. Then you can just pass any number of tests and avoid having to use an array. What do you think?

Reply:
Yeah, good point, I agree that that's nicer. It will require changing all the perf tests again though. So I'm not going to make the change right away, just in case there is more feedback on this API.